src/app/meshes/sources.ts

```diff
@@ -0,0 +1,17 @@
+import { DataSourceResponse } from '@/app/application/services/data-source/DataSource'
+import type KumaApi from '@/services/kuma-api/KumaApi'
+import type {
+  MeshInsight,
+} from '@/types/index.d'
+type MeshParams = {
+  mesh: string
+}
+export type MeshInsightSource = DataSourceResponse<MeshInsight>
+export const sources = (api: KumaApi) => {
+  return {
+    '/:mesh/insights': async (params: MeshParams, source: {close: () => void}) => {
+      source.close()
+      return api.getMeshInsights({ name: params.mesh })
```

Comment:
Wanted to note that to avoid too much change in this PR I'm just reusing `KumaApi` as a HTTP client. In the future we can slowly move away from a large single API client and towards 'module' specific "sources of data". We can eventually pass in a lighter `request` HTTP client into where we are currently passing `api: KumaApi`, but we can do that incrementally as tech debt work.

All that is required for a `Source` _right now_ is that it returns a Promise. That Promise could be generated by anything, `fetch`, a change to `localStorage` value, or a change to a users session. Currently, if you want it to poll, you just remove the `source.close` and let the source stay open keep 'requesting' (this is potentially a temporary way of polling to avoid adding too much at once)